File: C++/remove_linked_list_elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */

class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode* tmp = new ListNode(NULL);
        ListNode* prev = tmp;
        
        tmp->next = head;
        
        while(head != NULL) {
            if(head->val == val) {
                prev->next = head->next;
                head = head->next;
                continue;
            }
            prev = head;
            head = head->next;
        }
        return tmp->next;
    }
};

File: Python3/remove_linked_list_elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:    
        tmp = prev = ListNode(None)
        tmp.next = head
        
        while head:
            if head.val == val:
                prev.next, head = head.next, head.next
                continue
            prev, head = head, head.next
                
        return tmp.next
